### ISSUE TITLE
Multi-line binary expression in call args causes closing paren to be placed on same line as last arg

### ISSUE THREAD

**Issue by jakebailey** [dprint/dprint-plugin-typescript#550]
**Describe the bug**

dprint-plugin-typescript version: 0.86.1

**Input Code**

```ts
forEach(arr, f =>
    a()
    && b()
);

```

**Expected Output**

Maybe...

```ts
forEach(arr, f => a() && b());

forEach(
  arr,
  f => a() && b()
);
```

**Actual Output**

```ts
forEach(arr, f =>
  a()
  && b());
```

[Playground Link](https://dprint.dev/playground/#code/GYewTgoghgxgFgCimMAaABMdBeAfAWACh0T0oEBKI09AMlvQCNKiKBuIzwmEAOwGcALugAeOTDgLFS5KtJL0mlIA/language/typescript)

Truthfully, I'm not sure what the bug is here, but it seems a little odd to me in the first code snippet that the paren would get moved up a line.

See also: https://github.com/microsoft/TypeScript/pull/54820#discussion_r1246068978

**Comment by dsherret** [comment]
FWIW, you can manually fix this by moving the binary expression to be on one line or by moving the first parameter to the second line if you want it on multiple lines.

**Comment by jakebailey** [comment]
Indeed, I'm just trying to close the loop on all of the "oddities" on the TS PR before we merge it and forget about them.
